core/src/main/java/apoc/trigger/TriggerHandlerNewProcedures.java

```diff
@@ -47,59 +49,65 @@ public static void checkEnabled() {
     public static Map<String, Object> install(String databaseName, String triggerName, String statement, Map<String,Object> selector, Map<String,Object> params) {
         final HashMap<String, Object> previous = new HashMap<>();
 
+        final Node[] node = new Node[1];
         withSystemDb(tx -> {
-            Node node = Util.mergeNode(tx, SystemLabels.ApocTrigger, null,
+            node[0] = Util.mergeNode(tx, SystemLabels.ApocTrigger, null,
                     Pair.of(SystemPropertyKeys.database.name(), databaseName),
                     Pair.of(SystemPropertyKeys.name.name(), triggerName));
             
             // we'll return previous trigger info
-            previous.putAll(TriggerHandlerWrite.toTriggerInfo(node));
+            previous.putAll(toTriggerInfo(node[0]));
             
-            node.setProperty(SystemPropertyKeys.statement.name(), statement);
-            node.setProperty(SystemPropertyKeys.selector.name(), Util.toJson(selector));
-            node.setProperty(SystemPropertyKeys.params.name(), Util.toJson(params));
-            node.setProperty(SystemPropertyKeys.paused.name(), false);
+            node[0].setProperty(SystemPropertyKeys.statement.name(), statement);
+            node[0].setProperty(SystemPropertyKeys.selector.name(), Util.toJson(selector));
+            node[0].setProperty(SystemPropertyKeys.params.name(), Util.toJson(params));
+            node[0].setProperty(SystemPropertyKeys.paused.name(), false);
 
-            setLastUpdate(databaseName, tx);
             return null;
         });
 
+        setLastUpdate(databaseName, node[0]);
+
```

Comment:
Could you help me understand why this change is needed?

The way I see it, splitting the trigger creation/update logic in one transaction, and then update logic for the metadata in another transaction is harmful. Because the first transaction could commit and the second transaction could fail for whatever reason. This scenario would leave the dbms in an inconsistent state, and would mean that although a new triggers were installed in "system", the other databases would not update because they the metadata wouldn't update.

They should all happen in the same transaction (not separate and not nested transactions).

And to be even more correct, we should remove all the calls to withSystemDb which opens a new nested transaction which is discussed in this [comment thread](https://github.com/neo4j-contrib/neo4j-apoc-procedures/pull/3183#discussion_r1005585858) but we don't have to do that now.

Let's have a discussion Slack or a call if there is any confusion here.

Reply:
Maybe it's not needed (I hope), I should have found another way to solve that.
I created a new pr with the flaky stuff, https://github.com/neo4j-contrib/neo4j-apoc-procedures/pull/3355.